Doc/library/importlib.rst

```diff
@@ -1284,66 +1288,71 @@ find and load modules.
    A specification for a module's import-system-related state.  This is
    typically exposed as the module's ``__spec__`` attribute.  In the
    descriptions below, the names in parentheses give the corresponding
-   attribute available directly on the module object.
-   E.g. ``module.__spec__.origin == module.__file__``.  Note however that
-   while the *values* are usually equivalent, they can differ since there is
-   no synchronization between the two objects.  Thus it is possible to update
-   the module's ``__path__`` at runtime, and this will not be automatically
-   reflected in ``__spec__.submodule_search_locations``.
+   attribute available directly on the module object. For example,
+   ``module.__spec__.name == module.__name__``.  Note however that while
+   the *values* are usually equivalent, they can differ since there is no
+   synchronization between the two objects.  For example, it is possible
+   to update the module's ``__name__`` at runtime and this will not be
+   automatically reflected in the module's ``__spec__.name``, or the
+   other way round.
 
    .. versionadded:: 3.4
 
    .. attribute:: name
 
    (``__name__``)
 
-   A string for the fully-qualified name of the module.
+   The module's fully-qualified name.
 
    .. attribute:: loader
 
    (``__loader__``)
 
-   The loader to use for loading.  For namespace packages this should be
-   set to ``None``.
+   The module's loader.
 
    .. attribute:: origin
 
    (``__file__``)
 
-   Name of the place from which the module is loaded, e.g. "builtin" for
-   built-in modules and the filename for modules loaded from source.
-   Normally "origin" should be set, but it may be ``None`` (the default)
-   which indicates it is unspecified (e.g. for namespace packages).
+   The origin of the module's source code.
+   It is ``'built-in'`` for built-in modules, ``'frozen'`` for
+   frozen modules and ``None`` for namespace packages.
 
    .. attribute:: submodule_search_locations
 
    (``__path__``)
 
-   List of strings for where to find submodules, if a package (``None``
-   otherwise).
+   The (possibly empty) iterable  of the locations of the module's
+   submodules.
+   It is ``None`` for non-package modules.
 
    .. attribute:: loader_state
 
-   Container of extra module-specific data for use during loading (or
-   ``None``).
+   The container of the module's additional data meant for use by
```

Comment:
```
The finder may set this to an object containing additional data for the loader to use when loading the module.  Otherwise it will be ``None``.
```

Reply:
Okay.